### PR TITLE
Copyedits to Byterun Interpreter chapter

### PR DESCRIPTION
**Pull request by amyrbrown**
Here are my copyedits to the Byterun chapter. Nothing major, mostly just typo fixes and formatting changes. 

There are three queries in the PR, on (new) lines 144, 314, and 393.

### CODE REVIEW

File: interpreter/chapter.txt [hunk 1]
```diff
@@ -131,9 +137,10 @@ Remember that given valid instruction sets, without any changes to our interpret
         "numbers": [7, 5, 8] }
 ~~~~
 
-At this point, we can begin to see how this structure is extensible - we can add methods on the interpreter object that describe many more operations (as long as we have a compiler to hand us well-formed instruction sets).
+At this point, we can begin to see how this structure is extensible: we can add methods on the interpreter object that describe many more operations (as long as we have a compiler to hand us well-formed instruction sets).
+
+#### Variables
 
-### Variables
 Next let's add variables to our interpreter. Variables require four pieces: an instruction for storing the value of a variable, `STORE_NAME`; an instruction for retrieving it, `LOAD_NAME`; and a mapping from variable names to values. For now, we'll ignore namespaces and scoping, so we can store the variable mapping on the interpreter object itself. Finally, we'll have to make sure that `what_to_execute` has a list of the variable names, in addition to its list of constants.
```
Comment (amyrbrown) [inline interpreter/chapter.txt]: The text says "Variable require four pieces" but the follow list seems to only contain three things: 

- an instruction for storing,
- an instruction for retrieving, and
- a mapping from variable names to values.

Do we need one more thing on that list, or should we change it to "Variables require three pieces"?

File: interpreter/chapter.txt [hunk 2]
```diff
@@ -296,11 +307,11 @@ So far, the interpreter has executed code simply by stepping through the instruc
...
+Python loops also rely on jumping. In the bytecode below, notice that the line `while x < 5` generates almost identical bytecode to `if x < 10`. In both cases, the comparison is calculated and then `POP_JUMP_IF_FALSE` controls which instruction is executed next. At the end of line 4 --- the end of the loop's body --- the instruction `JUMP_ABSOLUTE` always sends the interpreter back to instruction 9 at the top of the loop. When x < 10 becomes false, then `POP_JUMP_IF_FALSE` jumps the interpreter past the end of the loop, to instruction 34. (The instructions `SETUP_LOOP` and `POP_BLOCK` are responsible for setting up and cleaning up the loop, and we'll touch on them later when we discuss blocks.)
```
Comment (amyrbrown) [inline interpreter/chapter.txt]: Here you say "we'll touch on [the instructions `SETUP_LOOP` and `POP_BLOCK`] later", but they are not specifically referred to in the block section below. Suggest either removing the phrase "and we'll touch on them later", or expanding block discussion below to explicitly refer to the instructions.

File: interpreter/chapter.txt [hunk 3]
```diff
@@ -372,23 +386,28 @@ c  |       (oldest)      | -> data stack: []
...
+When Ned Batchelder and I were working on Byterun, for a long time we had a significant error in our implementation. Instead of having one data stack on each frame, we had just one data stack on the entire virtual machine. We had dozens of tests made up of little snippets of Python code which we ran through Byterun and through the real Python interpreter to make sure the same thing happened in both interpreters. Nearly all of these tests were passing. The only thing we couldn't get working was generators. Finally, reading the CPython code more carefully, we realized the mistake.[FIXME: footnote: My thanks to Michael Arntzenius for his insight on this bug.] Moving a data stack onto each frame fixed the problem.
```
Comment (amyrbrown) [inline interpreter/chapter.txt]: Changed Artzenius to Arntzenius as per http://www.rntz.net/ -- I assume that's the same person? (!)

File: interpreter/chapter.txt [hunk 4]
```diff
@@ -1,39 +1,43 @@
...
 
-## A Python interpreter
-Before we begin, let's narrow down what we mean by "a Python interpreter." The word "interpreter" can be used in a variety of different ways when discussing Python. Sometimes "interpreter" refers to the Python REPL, the interactive prompt you get by typing just "python" from the command line. Sometimes people use "the Python interpreter" more or less interchangeably with "Python" to talk about executing Python code from start to finish. In this chapter, "interpreter" has a more narrow meaning: it's the last step in the process of executing a Python program.
+### A Python Interpreter
 
-Before the interpreter takes over, Python performs three other steps: lexing, parsing, and compiling. Together, these steps transform the programmer's source code from lines of text into structured _code objects_ containing instructions that the interpreter knows how to understand. The interpreter's job is then to take these code objects and follow the instructions.
```
Comment (amyrbrown) [inline interpreter/chapter.txt]: Moved this paragraph down below the discussion of interpreters so the paragraph addressing what "interpreter" means comes first.

File: interpreter/chapter.txt [hunk 5]
```diff
@@ -372,23 +386,28 @@ c  |       (oldest)      | -> data stack: []
 k   ---------------------
 ```
 
-At this point, the interpreter is in the middle of the function call to `bar`. There are three frames on the call stack: one for the module level, one for the function `foo`, and one for `bar`. Once `bar` returns, the frame associated with it is popped off of the call stack and discarded.
```
Comment (amyrbrown) [inline interpreter/chapter.txt]: "off of" is considered too informal for books (although "out of" is fine -- English!)

Comment (akaptur) [inline interpreter/chapter.txt]: Huh! "out of" doesn't make sense in the context of a stack, but "popped from the call stack" would be fine.

File: interpreter/chapter.txt (hunk 3 above)
Comment (akaptur) [inline interpreter/chapter.txt]: :heart: Thank you for this terrific catch

File: interpreter/chapter.txt (hunk 1 above)
Comment (akaptur) [inline interpreter/chapter.txt]: I think the list of variable names referenced in "finally" is the fourth thing, but let me check my version history to make sure. Thanks for the catch!

File: interpreter/chapter.txt (hunk 2 above)
Comment (akaptur) [inline interpreter/chapter.txt]: Thanks - I'll take a look and see which of these is the better course of action.

File: interpreter/chapter.txt (hunk 4 above)
Comment (akaptur) [inline interpreter/chapter.txt]: hmmm, let me look at this in context.

**Comment by akaptur** [comment]
Thanks for the great comments, @amyrbrown! I'll take a look at the ones that require more thought and get back to you as soon as possible.

File: interpreter/chapter.txt (hunk 5 above)
Comment (amyrbrown) [inline interpreter/chapter.txt]: Sorry -- my note was confusing. All we need to change here is to remove the "of", to make it "popped off the call stack".

File: interpreter/chapter.txt (hunk 3 above)
Comment (MichaelDiBernardo) [inline interpreter/chapter.txt]: + :heart:

**Comment by MichaelDiBernardo** [comment]
Hi,

I'm going to merge this, open a new branch, and address the open comments. 

I'll check with @akaptur before merging, but I'd also like to early-access publish this chapter soon, and I don't think the open comments are stop-ship :)

File: interpreter/chapter.txt (hunk 1 above)
Comment (MichaelDiBernardo) [inline interpreter/chapter.txt]: For the record, I think the fourth part was the list of variable names that `what_to_execute` requires, which is the last sentence in the paragraph.

**Pull request merged**